plugins/catalog/src/components/EntityLayout/EntityLayout.tsx

```diff
@@ -185,33 +193,63 @@ export const EntityLayout = (props: EntityLayoutProps) => {
   const { kind, namespace, name } = useRouteRefParams(entityRouteRef);
   const { entity, loading, error } = useAsyncEntity();
   const location = useLocation();
-  const routes = useElementFilter(
-    children,
-    elements =>
-      elements
-        .selectByComponentData({
-          key: dataKey,
-          withStrictError:
-            'Child of EntityLayout must be an EntityLayout.Route',
-        })
-        .getElements<EntityLayoutRouteProps>() // all nodes, element data, maintain structure or not?
-        .flatMap(({ props: elementProps }) => {
-          if (!entity) {
-            return [];
-          } else if (elementProps.if && !elementProps.if(entity)) {
+
+  const routeElements = useElementFilter(children, elements =>
```

Comment:
There's another option here that might be a bit trickier but waaaaaay more generic, which is to add support for the `permission` prop to `useElementFilter`. We could add it later too either way though

Reply:
How do you think this would look? Something like this?

```ts
  const routeElements = useElementFilter(children, elements =>
    elements
      .selectByComponentData({
        key: dataKey,
        withStrictError: 'Child of EntityLayout must be an EntityLayout.Route',
      })
      .withPermissionResource({ entity })
      .getElements<EntityLayoutRouteProps>(),
  );
```

Reply:
Perhaps, unless we just rely on the entity context instead